state/secretbackends.go

```diff
@@ -116,6 +125,62 @@ func (s *secretBackendsStorage) toSecretBackend(doc *secretBackendDoc) *secrets.
 	}
 }
 
+// UpdateSecretBackend updates a new secret backend.
+func (s *secretBackendsStorage) UpdateSecretBackend(p UpdateSecretBackendParams) error {
+	secretBackendsColl, closer := s.st.db().GetCollection(secretBackendsC)
+	defer closer()
+
+	var doc secretBackendDoc
+	err := secretBackendsColl.FindId(p.ID).One(&doc)
+	if err == mgo.ErrNotFound {
```

Comment:
should really use errors.Is rather than relying on equality.

Reply:
we use this in 105 other places - the mgo api returns ErrNotFound directly